compiler/src/dmd/parse.d

```diff
@@ -4388,6 +4374,23 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
         }
     }
 
+    /**
+     * Parse `align` or `align(n)`
```

Comment:
It doesn't check for the align keyword in this function, so it's really just parsing an optional parenthesized expression

Reply:
Added `assert(token.value == TOK.align_);`, also this will soon be `align`, `align(default)` or `align(n)` see #20587